backend/fms_core/services/sample_lineage.py

```diff
@@ -72,18 +72,57 @@ def create_sample_lineage_graph(sampleId: int) -> Tuple[List[Dict[str, Any]], Li
     
     return (nodes, edges, errors)
 
-def get_library_size_for_derived_sample(derived_sample_id: int) -> Optional[int]:
+def get_sample_source_from_derived_sample(child_sample_id: int, child_derived_sample_id: int) -> List[int]:
     """
-    Provides the latest measured fragment_size related to a derived_sample
+    Provides the sample id of the last aliquot of a sample that completed an experiment run before pooling 
```

Comment:
what's aliquot?

Reply:
It is a sub-sample created when the lab takes an amount of a given sample to create a new sample. It is a normal sample in any given way (this is just the name used when your sample is split from an existing sample i think).